beetsplug/alternatives.py

```diff
@@ -372,7 +443,20 @@ def update(self, create=None):
                     self.set_path(item, dest)
                 elif action == self.ADD:
                     print_(u'+{0}'.format(displayable_path(dest)))
-                    self.create_symlink(item)
+                    replace = False
+                    if (os.path.islink(syspath(dest))):
+                        replace = input_yn(u"There is a symlink at '{0}', "
+                            "but it wasn't created by beets-alternatives.\n"
+                            "Overwrite it? (y/n)"
+                                .format(displayable_path(dest)),
+                            require=False)
+                    elif (os.path.isfile(syspath(dest))):
+                        replace = input_yn(u"There is a file at '{0}'. "
+                            "Are you sure you wan't to replace it with a "
+                            "symlink for your alternative collection? (y/n)"
+                            .format(displayable_path(dest)),
+                            require=True)
+                    self.create_symlink(item, replace)
```

Comment:
It’s a great idea to give the user the ability to react to this case. However, there are good arguments for not going this far.

* This introduces inconsistencies between the behavior for regular externals and symlink views. For regular externals the program will abort if the destination file exists and was not created by this plugin.
* The feature adds complexity.
* The improvement is only marginal. Without this the user will see an error message that the link cannot be created because the file already exists. They can then easily fix this themselves.

Reply:
Ok, I mainly implemented this because I thought that @dosoe 's library was broken in a way that would be fixed up like this. In the end, that probably wasn't the case. Do you think I should just remove these checks again (such that we abort/crash) in these cases, or keep them but only issue a warning (and continue processing)? I'll have to check again whether this is really a state that a symlink view can end up in after a crash.

Reply:
For this PP I’d prefer to keep the behavior consistent with the other external collections and just abort with an error. We should discuss the change in a separate issue and then implement it for both collections.

Reply:
> We should discuss the change in a separate issue and then implement it for both collections.

See #52